Author: Sergei Golovan
Description: Patch adds trivial processing for BIO_CTRL_GET_KTLS_SEND
 and BIO_CTRL_GET_KTLS_RECV control commands to make tcltls working
 with libssl 3.0.
Last-Modified: Sun, 20 Mar 2022 15:08:23 +0300
Debian-Bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1006587

--- a/tlsBIO.c
+++ b/tlsBIO.c
@@ -282,6 +282,18 @@
 			ret = ((Tcl_WriteRaw(chan, "", 0) >= 0) ? 1 : -1);
 			dprintf("BIO_CTRL_FLUSH returning value %li", ret);
 			break;
+#ifdef BIO_CTRL_GET_KTLS_SEND
+		case BIO_CTRL_GET_KTLS_SEND:
+			dprintf("Got BIO_CTRL_GET_KTLS_SEND");
+			ret = 0;
+			break;
+#endif
+#ifdef BIO_CTRL_GET_KTLS_RECV
+		case BIO_CTRL_GET_KTLS_RECV:
+			dprintf("Got BIO_CTRL_GET_KTLS_RECV");
+			ret = 0;
+			break;
+#endif
 		default:
 			dprintf("Got unknown control command (%i)", cmd);
 			ret = -2;
